tests/test_unit/test_vector.py

```diff
@@ -121,3 +121,54 @@ def test_pol2cart(self, ds, expected_exception, request):
         with expected_exception:
             result = vector.pol2cart(ds.pol)
             xr.testing.assert_allclose(result, ds.cart)
+
+    @pytest.mark.parametrize(
+        "ds, expected_exception",
+        [
+            ("cart_pol_dataset", does_not_raise()),
+            ("cart_pol_dataset_with_nan", does_not_raise()),
+            ("cart_pol_dataset_missing_cart_dim", pytest.raises(ValueError)),
+            (
+                "cart_pol_dataset_missing_cart_coords",
+                pytest.raises(ValueError),
+            ),
+        ],
+    )
+    def test_magnitude(self, ds, expected_exception, request):
+        """Test vector magnitude with known values."""
+        ds = request.getfixturevalue(ds)
+        with expected_exception:
+            result = vector.magnitude(ds.cart)
+            expected = np.sqrt(
+                ds.cart.sel(space="x") ** 2 + ds.cart.sel(space="y") ** 2
+            )
+            xr.testing.assert_allclose(result, expected)
+            # result should be the same from Cartesian and polar coordinates
+            xr.testing.assert_allclose(result, vector.magnitude(ds.pol))
+            # The result should only contain the time dimension.
+            assert result.dims == ("time",)
+
+    @pytest.mark.parametrize(
+        "ds, expected_exception",
+        [
+            ("cart_pol_dataset", does_not_raise()),
+            ("cart_pol_dataset_with_nan", does_not_raise()),
+            ("cart_pol_dataset_missing_cart_dim", pytest.raises(ValueError)),
+        ],
+    )
+    def test_normalize(self, ds, expected_exception, request):
+        """Test data normalization (division by magnitude)."""
+        ds = request.getfixturevalue(ds)
+        with expected_exception:
+            normalized = vector.normalize(ds.cart)
+            # the normalized data should have the same dimensions as the input
+            assert normalized.dims == ds.cart.dims
+            # the first time point is NaN because the input vector is [0, 0]
+            # (zero-division during normalization).
+            assert normalized.sel(time=0).isnull().all()
+            # the magnitude of the normalized vector should be 1 for all
+            # time points except for the expected NaNs.
+            normalized_mag = vector.magnitude(normalized).values
+            expected_mag = np.ones_like(normalized_mag)
+            expected_mag[normalized.isnull().any("space")] = np.nan
+            np.testing.assert_allclose(normalized_mag, expected_mag)
```

Comment:
I like the comments before the assertions! But I find it easier to read this test if we add some spaces before these comments, to separate the assertions too. I feel this would help external contributors too.